src/sharing/components/Recipient.jsx

```diff
@@ -35,34 +36,55 @@ export const UserAvatar = ({ name, url }) => (
   </div>
 )
 
-const Recipient = ({ t, contact, status, documentType, onUnshare }) => {
-  const name = getPrimaryEmail(contact)
-  const url = getPrimaryCozy(contact)
-  return (
-    <div className={styles['pho-recipient']}>
-      <Avatar name={name} />
-      <Identity name={name} url={url} />
-      {status && (
+class Recipient extends Component {
+  state = {
+    revoking: false
+  }
+
+  onUnshare = () => {
+    this.setState({ revoking: true })
+    this.props
+      .onUnshare(this.props.contact)
+      .then(() => this.setState({ revoking: false }))
```

Comment:
As the `setState` is used in an asynchronous function, we should use, as always, the function version of `setState`:

```js
.then(() => this.setState(state => ({ revoking: false })))
```

Reply:
Yep!

Reply:
I don't understand what difference it makes here. Whether you call `setState` with `{ revoking: false }` or `state => ({ revoking: false })`, the result will always be the same.

Reply:
I don't understand what difference it makes here. Whether you call `setState` with `{ revoking: false }` or `state => ({ revoking: false })`, the result will always be the same.